v8/flexsearch_plugin/README.md

```diff
@@ -0,0 +1,226 @@
+
+This js allows to click the button or press enter to search
+
+```html
+<script src="https://rawcdn.githack.com/nextapps-de/flexsearch/0.7.31/dist/flexsearch.bundle.js"></script>
+
+<script>
+document.addEventListener('DOMContentLoaded', function() {
+    var searchIndex = new FlexSearch.Index();  // Initialize FlexSearch
+    var index = {};  // This will store the index data globally within this script block
+
+    // Fetch the generated JSON file
+    fetch('search_index.json')
+    .then(response => response.json())
+    .then(data => {
+        index = data;
+        for (var key in index) {
+            if (index.hasOwnProperty(key)) {
+                searchIndex.add(key, index[key].content);
+            }
+        }
+    });
+
+    var input = document.getElementById('search_input');
+    var button = document.getElementById('search_button');
+
+    // Function to perform search
+    function performSearch() {
+        var query = input.value;
+        var results = searchIndex.search(query);
+        var resultsContainer = document.getElementById('search_results');
+        resultsContainer.innerHTML = ''; // Clear previous results
+        results.forEach(function(result) {
+            var div = document.createElement('div');
+            var link = document.createElement('a');
+            link.href = index[result].url;
+            link.textContent = index[result].title;
+            div.appendChild(link);
+            resultsContainer.appendChild(div);
+        });
+        document.getElementById('search_overlay').style.display = 'flex'; // Show the overlay
+    }
+
+    // Event listener for search button click
+    button.addEventListener('click', performSearch);
+
+    // Event listener for pressing enter key in the search input
+    input.addEventListener('keypress', function(event) {
+        if (event.key === "Enter" || event.keyCode === 13) {
+            event.preventDefault();  // Prevent the form from being submitted
+            performSearch();
+        }
+    });
+});
+
+// Function to close the search overlay
+function closeSearch() {
+    document.getElementById('search_overlay').style.display = 'none';
+}
+</script>
+```
+
+add this to your template where you want the search to appear
+
+```html
+<div id="search_overlay" style="display:none;">
+    <div id="search_content">
+        <button onclick="closeSearch()">Close</button>
+        <div id="search_results"></div>
+    </div>
+</div>
+<input type="text" id="search_input">
+<button id="search_button">Search</button>
+```
+
+add css for the overlay
+
+```html
+/* flexsearch_plugin*/
+
+#search_overlay {
+    position: fixed; /* Fixed position to cover the whole screen */
+    width: 100%;
+    height: 100%;
+    top: 0;
+    left: 0;
+    background: rgba(0, 0, 0, 0.8); /* Semi-transparent background */
+    z-index: 1000; /* Make sure it's on top of other content */
+    display: flex;
+    justify-content: center;
+    align-items: center;
```

Comment:
Multiple centered lines don’t look too great, consider removing the centering, and adding `<li>` to make the separate items stand out.

Reply:
Hi @Kwpolska  Thank you for your feedback. Indeed, it does look much better aligned to the left.

I made the suggested changes and now the results are in an ul list, aligned to left in the container. The overlay is still centered, though.

I updated the example image, added more css for the new ul and li elements and added a background on hoover for the li result elements.

![example_overlay](https://github.com/getnikola/plugins/assets/557703/56b00628-1e4e-461c-bb5f-498c3105a73f)